docs/src/configuration/services/opensearch.md

```diff
@@ -0,0 +1,167 @@
+---
+title: "OpenSearch (search service)"
+weight: 1
+description: |
+  OpenSearch is a distributed RESTful search engine built for the cloud.
+sidebarTitle: "OpenSearch"
+---
+
+{{< description >}}
+
+See the [OpenSearch documentation](https://opensearch.org/docs/1.2/) for more information.
+
+To switch from Elasticsearch, follow the same procedure as for [upgrading](#upgrading).
+
+## Supported versions
+
+<!--
+To update the versions in this table, use docs/data/registry.json
+-->
+| **Grid** | **Dedicated** | **Dedicated Generation 3** |
+|----------------------------------|---------------|---------------|
+|  {{< image-versions image="opensearch" status="supported" environment="grid" >}} | {{< image-versions image="opensearch" status="supported" environment="dedicated" >}} | {{< image-versions image="opensearch" status="supported" environment="dedicated-gen-3" >}} |
+
+## Relationship
+
+The format exposed in the `$PLATFORM_RELATIONSHIPS` [environment variable](../../development/variables.md#platformsh-provided-variables):
+
+{{< relationship "opensearch" >}}
+
+## Usage example
+
+{{% endpoint-description type="opensearch" noApp=true %}}
+
+[Service definition](./_index.md):
+
+{{< readFile file="src/registry/images/examples/full/opensearch.services.yaml" highlight="yaml" location=".platform/services.yaml" >}}
+
+[App configuration](../app/app-reference.md):
+
+{{< readFile file="src/registry/images/examples/full/opensearch.app.yaml" highlight="yaml" location=".platform.app.yaml" >}}
+
+{{% /endpoint-description %}}
+
+{{< note >}}
+
+When you create an index on OpenSearch,
+don't specify the `number_of_shards` or `number_of_replicas` settings in your OpenSearch API call.
+These values are set automatically based on available resources.
+
+{{< /note >}}
+
+## Authentication
+
+By default, OpenSearch has no authentication.
+No username or password is required to connect to it.
+
+You may optionally enable HTTP Basic authentication.
+To do so, include the following in your `services.yaml` configuration:
+
+```yaml {location=".platform/services.yaml"}
+search:
+    type: opensearch:1.2
+    disk: 2048
+    configuration:
+        authentication:
+            enabled: true
+```
+
+That enables mandatory HTTP Basic auth on all requests.
+The credentials are available in any relationships that point at that service,
+in the `username` and `password` properties.
+
+This functionality is generally not required if OpenSearch isn't exposed on its own public HTTP route.
+However, certain applications may require it, or it allows you to safely expose OpenSearch directly to the web.
+To do so, add a route to `routes.yaml` that has `search:opensearch` as its upstream
+(where `search` is whatever you named the service in `services.yaml`).
+For example:
+
+```yaml {location=".platform/routes.yaml"}
+"https://os.{default}":
+    type: upstream
+    upstream: search:opensearch
+```
+
```

Comment:
Maybe we could add a note on [how to add basic auth](https://docs.platform.sh/administration/web/configure-environment.html#http-access-control) for an environment (bot both this and the ES guides)?